doxia-integration-tools/src/main/java/org/apache/maven/doxia/tools/DefaultSiteTool.java

```diff
@@ -560,39 +559,13 @@ private void populateModulesMenu(
         }
 
         // we require child modules and reactors to process module menu
-        if (project.getModules().size() > 0) {
+        List<MavenProject> projectModules = project.getCollectedProjects();
+        if (projectModules != null && !projectModules.isEmpty()) {
             if (menu.getName() == null) {
                 menu.setName(i18n.getString("site-tool", locale, "decorationModel.menu.projectmodules"));
             }
 
-            for (String module : (List<String>) project.getModules()) {
-                MavenProject moduleProject = getModuleFromReactor(project, reactorProjects, module);
-
-                if (moduleProject == null) {
-                    LOGGER.warn("Module " + module + " not found in reactor: loading locally");
-
-                    File f = new File(project.getBasedir(), module + "/pom.xml");
-                    if (f.exists()) {
-                        try {
-                            ProjectBuildingRequest request = new DefaultProjectBuildingRequest();
-                            request.setLocalRepository(localRepository);
-
-                            ProjectBuildingResult result = projectBuilder.build(f, request);
-                            moduleProject = result.getProject();
-                        } catch (ProjectBuildingException e) {
-                            throw new SiteToolException("Unable to read local module POM", e);
-                        }
-                    } else {
-                        LOGGER.warn("No filesystem module POM available");
-
-                        moduleProject = new MavenProject();
-                        moduleProject.setName(module);
-                        moduleProject.setDistributionManagement(new DistributionManagement());
-                        moduleProject.getDistributionManagement().setSite(new Site());
-                        moduleProject.getDistributionManagement().getSite().setUrl(module);
-                    }
-                }
-
```

Comment:
The removal of this block causes the menu items to disappear when, e.g., `mvn site -N` is executed...which isn't necessarily wrong. @gnodet @hboutemy WDYT? It be consistent with the remaining Maven behavior.

Reply:
Seems to work now with a624f1a5d113377f98a67eb698d8bbeffad4ce84.